### PR TITLE
HDDS-9804. ozone freon dfsg does not work with relative --path

### PR DESCRIPTION
**Pull request by jojochuang**
## What changes were proposed in this pull request?
Add check to prevent invalid --path parameter in ozone freon dfsg command.

Please describe your PR in detail:
* It's a trivial change but took me several hours to troubleshoot so thought it's worth fixing.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-9804

## How was this patch tested?

Manually tested on a cluster.

### CODE REVIEW

**Review by nandakumar131 — APPROVED** [review]
+1, LGTM.

File: hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/HadoopFsGenerator.java [hunk 1]
```diff
@@ -87,6 +87,10 @@ public Void call() throws Exception {
 
     configuration = createOzoneConfiguration();
     uri = URI.create(rootPath);
+    String scheme = uri.getScheme();
+    if (scheme == null) {
+      throw new IllegalArgumentException("--path requires FQDN");
+    }
```
Comment (ayushtkn) [inline hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/HadoopFsGenerator.java:93]: Shouldn't we use the default FS in that case? If no FS is specified explicitly, the defaultFs should be used, right? Something like this:
```
    if (scheme == null) {
      URI defaultUri = FileSystem.getDefaultUri(configuration)
      uri = new URI(defaultUri.getScheme(), defaultUri.getAuthority(), uri.getPath());
    }
```

File: hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/HadoopFsGenerator.java [hunk 2]
```diff
@@ -87,6 +87,10 @@ public Void call() throws Exception {
 
     configuration = createOzoneConfiguration();
     uri = URI.create(rootPath);
+    String scheme = uri.getScheme();
+    if (scheme == null) {
```
Comment (ayushtkn) [inline hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/HadoopFsGenerator.java:91]: does this work?
```
    if (!uri.isAbsolute()) {
```

File: hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/HadoopFsGenerator.java [hunk 3]
```diff
@@ -87,6 +87,10 @@ public Void call() throws Exception {
...
+      throw new IllegalArgumentException("--path requires FQDN");
+    }
     String disableCacheName = String.format("fs.%s.impl.disable.cache",
         uri.getScheme());
```
[the remark this answers is not in the document]
Comment (adoroszlai) [reply hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/HadoopFsGenerator.java:95]: Maybe should also fail if `scheme` is `file`.

**Review by adoroszlai — APPROVED** [review]
Thanks @jojochuang for updating the patch.

**Review by ayushtkn — APPROVED** [review]
LGTM

**Comment by adoroszlai** [comment]
Thanks @jojochuang for the patch, @ayushtkn, @nandakumar131 for the review.

**Pull request merged**